core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -419,6 +419,94 @@ public static int getFirstIndexUsingBinarySearch(FixedLengthDimensionDataChunk d
     return -(low + 1);
   }
 
+  public static int[] getRangeIndexUsingBinarySearch(
```

Comment:
There is not much difference between `getFirstIndexUsingBinarySearch` and this method,  I remembered in your last PR you have done binary search even for getting the ranges, what happened to it, did you get any functional or performance issues?